### PR TITLE
Methods and tests for winningest_coach and worst_coach.

### PR DESCRIPTION
**Pull request by lambo1986**
fix: winningest_coach and worst_coach method tests changed to match datasets

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -97,6 +97,68 @@ def highest_scoring_home_team
     find_team(:home, "max")
   end
 
+  def winningest_coach(season)
```
Comment (BarryA) [inline lib/stat_tracker.rb:100]: Is winningest a word? :satisfied:

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -97,6 +97,68 @@ def highest_scoring_home_team
...
+      games_total_by_coach[game.head_coach] += 1 if game.result != nil
+    end
+    games_total_by_coach.each do |coach, points|
+      coach_win_hash.each do |c, p|
+        coach_win_percentage[c] = p.fdiv(points).round(2) if coach == c
+      end
+    end
+    coach_win_percentage.select do |coach, win_percent|
+      winner = coach if win_percent == coach_win_percentage.values.max #&&
+      # coach_win_hash.values.max == coach_win_hash[coach] #only works for small dataset
```
Comment (yainporter) [inline lib/stat_tracker.rb]: Hey, why is this part commented out?

Comment (lambo1986) [reply lib/stat_tracker.rb]: I wasn't using that part for a certain set of data. but I think I need it now!

Comment (lambo1986) [reply lib/stat_tracker.rb]: I fixed it!

**Review by yainporter — APPROVED** [review]
Awesome, thanks for fixing that method, Nathan, glad it's working correctly now!

**Pull request merged**